Tasks/VsTest/distributedtest.ts

```diff
@@ -39,27 +39,9 @@ export class DistributedTest {
         }
     }
 
-    private async registerAndConfigureAgent() {
-        tl.debug('Configure the Agent with DTA... Invoking the createAgent REST API');
-
+    private async beginDtaExecutionHost() {
         try {
-            const agentId = await ta.TestAgent.createAgent(this.dtaTestConfig.dtaEnvironment, 3);
-            ci.publishEvent({
-                environmenturi: this.dtaTestConfig.dtaEnvironment.environmentUri, agentid: agentId,
-                agentsize: this.dtaTestConfig.numberOfAgentsInPhase, vsTestConsole: this.dtaTestConfig.useVsTestConsole,
-                batchsize: this.dtaTestConfig.numberOfTestCasesPerSlice
-            });
-
-            await this.startDtaExecutionHost(agentId);
-            await this.startDtaTestRun();
-            try {
-                if (this.dtaPid !== -1) {
-                    tl.debug('Trying to kill the Modules/DTAExecutionHost.exe process with pid :' + this.dtaPid);
-                    process.kill(this.dtaPid);
-                }
-            } catch (error) {
-                tl.warning('Modules/DTAExecutionHost.exe process kill failed, pid: ' + this.dtaPid + ' , error :' + error);
-            }
+            await this.startDtaExecutionHost();
             tl.setResult(tl.TaskResult.Succeeded, 'Task succeeded');
```

Comment:
if agent registration fails, you are retuning exit code 1. but where we are marking the task as failed?

Reply:
Modified the code to address this issue